core/service/manager.go

```diff
@@ -18,47 +18,64 @@
 package service
 
 import (
-	"crypto/x509/pkix"
-
 	log "github.com/cihub/seelog"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/state"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn/tls"
 	"github.com/mysteriumnetwork/node/communication"
-	"github.com/mysteriumnetwork/node/core/ip"
-	"github.com/mysteriumnetwork/node/core/location"
+	nats_dialog "github.com/mysteriumnetwork/node/communication/nats/dialog"
+	nats_discovery "github.com/mysteriumnetwork/node/communication/nats/discovery"
 	"github.com/mysteriumnetwork/node/discovery"
 	"github.com/mysteriumnetwork/node/identity"
+	identity_registry "github.com/mysteriumnetwork/node/identity/registry"
 	identity_selector "github.com/mysteriumnetwork/node/identity/selector"
+	"github.com/mysteriumnetwork/node/logconfig"
 	"github.com/mysteriumnetwork/node/metadata"
-	"github.com/mysteriumnetwork/node/nat"
 	dto_discovery "github.com/mysteriumnetwork/node/service_discovery/dto"
-	openvpn_discovery "github.com/mysteriumnetwork/node/services/openvpn/discovery"
-	"github.com/mysteriumnetwork/node/session"
 )
 
-// Manager represent entrypoint for Mysterium service with top level components
+const logPrefix = "[service-manager] "
+
+// Service interface represents pluggable Mysterium service
+type Service interface {
+	Start(providerID identity.Identity, dialogWaiter communication.DialogWaiter) (dto_discovery.ServiceProposal, error)
```

Comment:
That's strange that service depends on dialog waiter. Also does the same instance should be started with separate dialog waiters? This dependency looks more like implementation specific to me, not by service interface itself

Reply:
Currently did not know how to go. So I postponed to more clear interface, then we have more 2 services later